pyrsa/data/dataset.py

```diff
@@ -0,0 +1,165 @@
+                        from obs dimension
+            value:      the value by which the subset selection is made
+                        from obs dimension
+
+        Returns:
+            Dataset, with subset defined by the selected obs_descriptor
+        """
+        raise NotImplementedError(
+            "subset_obs function not implemented in used Dataset class!"
+        )
+
+    def subset_channel(self, by, value):
+        """ Returns a subsetted Dataset defined by certain channel value
+        Args:
+            by(String): the descriptor by which the subset selection is made
+                        from channel dimension
+            value:      the value by which the subset selection is made
+                        from channel dimension
+
+        Returns:
+            Dataset, with subset defined by the selected channel_descriptor
+        """
+        raise NotImplementedError(
+            "subset_channel function not implemented in used Dataset class!"
+        )
+
+
+class Dataset(DatasetBase):
+    """
+    Dataset class is a standard version of DatasetBase.
+    It contains one data set - or multiple data sets with the same structure
+    """
+    def split_obs(self, by):
+        """ Returns a list Datasets splited by obs
+        Args:
+            by(String): the descriptor by which the splitting is made
+
+        Returns:
+            list of Datasets, splitted by the selected obs_descriptor
+        """
+        unique_values = set(self.obs_descriptors[by])
+        dataset_list = []
+        for v in unique_values:
+            dataset_list.append(
+                self.measurements[:, self.obs_descriptors[by] == v, :])
+        return dataset_list
+        # TODO: for 3d measurements, need implementations.
+
+    def split_channel(self, by):
+        """ Returns a list Datasets splited by channels
+        Args:
+            by(String): the descriptor by which the splitting is made
+
+        Returns:
+            list of Datasets,  splitted by the selected channel_descriptor
+        """
+        unique_values = set(self.channel_descriptors[by])
+        dataset_list = []
+        for v in unique_values:
+            dataset_list.append(
+                self.measurements[:, :, self.channel_descriptors[by] == v])
+        return dataset_list
+        # TODO: for 3d measurements, need implementations.
+
+    def subset_obs(self, by, value):
+        """ Returns a subsetted Dataset defined by certain obs value
+        Args:
+            by(String): the descriptor by which the subset selection
+                        is made from obs dimension
+            value:      the value by which the subset selection is made
+                        from obs dimension
+
+        Returns:
+            Dataset, with subset defined by the selected obs_descriptor
+        """
+        return self.measurements[:, self.obs_descriptors[by] == value, :]
+        # TODO: for 3d measurements, need implementations.
```

Comment:
please convert the *TODO* comments to an issue before merging. I think the master branch should not contain *TODOs*.

Reply:
Thanks for the note! Will clean up and push next week.

Reply:
cleaned up and pushed.